user-manager/src/main/java/com/qpay/usermanager/service/impl/CustomerServiceImpl.java

```diff
@@ -1,64 +1,83 @@
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
+@Transactional(isolation = Isolation.READ_COMMITTED)
 public class CustomerServiceImpl implements CustomerService {
 
     private final CustomerRepository customerRepository;
 
     private final CustomerMapper customerMapper;
 
+    private final UserCredentialsMapper userCredentialsMapper;
+
     private final MerchantRepository merchantRepository;
 
+    private final AuthenticationClient authenticationClient;
+
     public CustomerEntity getCustomerById(final long id) {
         return customerRepository.findById(id).orElseThrow(CustomerNotFoundException::new);
     }
 
-    public CustomerEntity addCustomer(final CustomerModification customerModification) {
-        existsMerchantByEmail(customerModification.email());
-
-        var customerEntity = customerMapper.map(customerModification);
+    public CustomerEntity addCustomer(final CustomerCreation customerCreation) {
+        existsMerchantByEmail(customerCreation.email());
+        var customerEntity = customerMapper.map(customerCreation);
         try {
             customerEntity = customerRepository.save(customerEntity);
         } catch (final DataIntegrityViolationException e) {
             log.error(e.getMessage());
-            throw new CustomerAlreadyExistsException(e.getMessage());
+            throw new UserAlreadyExistsException(e.getMessage());
         }
+        final var newUserCredentials  = userCredentialsMapper.mapCustomerCreation(customerCreation);
+        authenticationClient.createUser(newUserCredentials );
         return customerEntity;
     }
 
     public CustomerEntity updateCustomer(final CustomerModification customerModification, final long id) {
         existsMerchantByEmail(customerModification.email());
-
         final var customerEntity = getCustomerById(id);
+        final var previousEmail = customerEntity.getEmail();
         customerEntity.setEmail(customerModification.email());
         customerEntity.setName(customerModification.name());
         customerEntity.setPassword(customerModification.password());
-        return customerRepository.save(customerEntity);
+        final var savedCustomerEntity = customerRepository.save(customerEntity);
+        sendAuthenticationUserModification(customerModification, previousEmail);
+        return savedCustomerEntity;
     }
 
     public void deleteCustomer(final long id) {
+        final var customerEntity = getCustomerById(id);
         customerRepository.deleteById(id);
+        authenticationClient.deleteUser(customerEntity.getEmail());
+    }
+
+    private void sendAuthenticationUserModification(final CustomerModification customerModification, final String previousEmail) {
+        final var userCredentialsModification = userCredentialsMapper.mapCustomerModification(customerModification);
+        authenticationClient.updateUser(previousEmail, userCredentialsModification);
```

Comment:
You don't update user. Previous time I wrote create user and so on as an example. You have to adapt it yourself for a particular case. You update credentials